### PR TITLE
Refactor SecurityToken ops

### PR DESCRIPTION
**Pull request by Valodim**
While investigating #2179 and #2049, I looked at how our decryption code worked in SecurityTokenConnection and noticed there were a couple of points I didn't understand and that weren't documented - so I wrote unit tests for them, and refactored until I knew what was going on :)

This PR now does a few things:
- extract particular use cases from SecurityTokenConnection into their own UseCase classes
- add unit tests for decryption on token
- refactor the decryption use case a bit for readability (this is kind of a WIP, not sure if I want to spend more time)
- refactor SecurityTokenConnection, in particular split up and reorder some methods

I also added a test case against that compares what we currently emit against ee8cd3862f65de580ed949bc838628610e22cd98. This is not currently identical, but could be made so if we find out it's more compatible (see #2049). But I read the openpgp card implementation code, and can't figure out any reason why this wouldn't work, so it's a bit of poking in the dark.

All of these are true refactorings - there should be no semantic changes in here.

### CODE REVIEW

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/GenerateKeyUseCase.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package org.sufficientlysecure.keychain.securitytoken;
+
+
+import java.io.IOException;
+
+import org.sufficientlysecure.keychain.util.Passphrase;
+
+
+public class GenerateKeyUseCase {
```
Comment (nmikhailov) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/GenerateKeyUseCase.java]: Looks like this is not used anywhere

Comment (Valodim) [reply OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/GenerateKeyUseCase.java]: It's not, never was iirc. I guess we left it in just to keep the code close :)

**Comment by dschuermann** [comment]
Please add license headers

**Comment by Valodim** [comment]
done :+1:

**Pull request merged**